processor/src/trace/mod.rs

```diff
@@ -86,7 +87,7 @@ impl ExecutionTrace {
         Self {
             meta: Vec::new(),
             layout: TraceLayout::new(TRACE_WIDTH, [AUX_TRACE_WIDTH], [AUX_TRACE_RAND_ELEMENTS]),
-            main_trace: ColMatrix::new(main_trace),
+            main_trace: MainTrace::new(ColMatrix::new(main_trace)),
```

Comment:
`finalize_trace()` should return a `MainTrace` instead of `Vec<Vec<Felt>>` in the first field of the tuple